jobs/build-podman-os.Jenkinsfile

```diff
@@ -0,0 +1,360 @@
+def gitref, commit, shortcommit
+
+def artifacts = [
+    [platform: "qemu", suffix: "qcow2"],
+    [platform: "applehv", suffix: "raw"],
+    [platform: "hyperv", suffix: "vhdx"]
+]
+
+node {
+    checkout scm
+    // these are script global vars
+    pipeutils = load("utils.groovy")
+}
+
+// Keep it here for now instead of utils
+// since it is podman job only
+
+def generate_disk_images(artifacts, shortcommit, arch, repo) {
+    generate_diskvar_json(shortcommit, arch, artifacts, repo)
+    // First create the qcow2 and store it in cache
+    supermin_run("--cache", arch, "qcow2", shortcommit)
+    def artifacts_runvm = artifacts.drop(1) // drop qcow2
+    parallel artifacts_runvm.collectEntries{artifact -> [artifact["platform"], {
+        supermin_run("--snapshot", arch, "${artifact["suffix"]}", shortcommit)
+    }]}
+}
+
+def generate_diskvar_json(shortcommit, arch, artifacts, repo) {
+    artifacts.each { artifact ->
+        def jsonContent = """
+        {
+            "osname": "fedora-coreos",
+            "container-repo": "$repo",
+            "container-tag": "${arch}-${shortcommit}",
+            "image-type": "${artifact["platform"]}",
+            "container-imgref": "ostree-remote-registry:fedora:quay.io/podman/machine-images:5.0",
```

Comment:
if we do https://github.com/coreos/fedora-coreos-pipeline/pull/968/files#r1531138529 then this will need updating too